fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -4221,7 +4221,7 @@ public DataType visitPrimitiveDataType(PrimitiveDataTypeContext ctx) {
             }
             List<String> l = Lists.newArrayList(dataType);
             ctx.INTEGER_VALUE().stream().map(ParseTree::getText).forEach(l::add);
-            return DataType.convertPrimitiveFromStrings(l);
+            return DataType.convertPrimitiveFromStrings(l).conversion();
```

Comment:
should not change here, like ArrayType, u should add conversion in AggState class constructor